core/src/main/scala/org/apache/spark/SparkException.scala

```diff
@@ -43,3 +43,9 @@ private[spark] case class SparkUserAppException(exitCode: Int)
  */
 private[spark] case class ExecutorDeadException(message: String)
   extends SparkException(message)
+
+/**
+ * Exception thrown when Spark returns different result after upgrading to a new version.
+ */
+private[spark] class SparkUpgradeException(version: String, message: String, cause: Throwable)
+  extends SparkException(s"Exception for upgrading to Spark $version: $message", cause)
```

Comment:
`You may get a different result due to the upgrading of Spark $version: $message`